src/enzo/DetermineSEDParameters.C

```diff
@@ -0,0 +1,290 @@
+			 {2.883229e+57, 8.072469e-01, 2.280500e-02, 8.348202e-02, 4.014262e-04, 1.684695e-04},
+			 {1.746463e+58, 6.558310e-01, 2.931956e-02, 1.960125e-01, 6.487986e-03, 2.781257e-04},
+			 {1.016622e+59, 4.899245e-01, 3.021142e-02, 2.987737e-01, 5.118229e-02, 4.777958e-04}};
+		
+static int CalculateArrayIndex(float Mass, float AccRate);	
+extern "C" void FORTRAN_NAME(stellar)(float *smdot, float *dt_enzo, float *slum, float *rad);
+	
+int DetermineSEDParameters(ActiveParticleType_SmartStar *SS, FLOAT Time, FLOAT dx)
+{
+
+  if((SmartStarBHRadiativeFeedback == FALSE) && (SmartStarStellarRadiativeFeedback == FALSE)) {
+      for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+	SS->RadiationEnergyBins[bin] = 0.0;
+	SS->RadiationSED[bin] = 0.0;
+      }
+      return SUCCESS;
+    }
+  float DensityUnits, LengthUnits, TemperatureUnits, TimeUnits,
+    VelocityUnits;
+  double MassUnits;
+  GetUnits(&DensityUnits, &LengthUnits, &TemperatureUnits,
+	   &TimeUnits, &VelocityUnits, Time);
+  MassUnits = DensityUnits * POW(LengthUnits,3);
+  double MassConversion = (float) (dx*dx*dx * double(MassUnits));  //convert to g from a density
+  double ParticleMass =  SS->ReturnMass()*MassConversion/SolarMass; //In solar masses
+
+#if STELLAR
+  static float ptime=0.0;
+  if(SS->ParticleClass == POPIII || SS->ParticleClass == SMS) {
+    SS->RadiationLifetime = 1e6*3.154e7/TimeUnits; //Code Time
+    SS->LuminosityPerSolarMass = 6.696798e49/40.0; //In physical units
+
+    float smdot, dt_enzo, slum, rad;
+    smdot=SS->AccretionRate[SS->TimeIndex]*MassUnits/TimeUnits;
+    dt_enzo=Time-ptime;
+
+  
+
+    if(ptime>0.0){
+      FORTRAN_NAME(stellar)(&smdot, &dt_enzo, &slum, &rad);
+      SS->LuminosityPerSolarMass = slum/ParticleMass;
+    }
+    else{
+      SS->LuminosityPerSolarMass = 1e10;
+    }
+  }
+  ptime = Time;
+  return SUCCESS;
+#endif
+  /* 
+   * The PopIII values are taken from Schaerer et al. 2002 Table 4.
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  if(SS->ParticleClass == POPIII) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Time
+    SS->LuminosityPerSolarMass = 6.696798e49/40.0; //In physical units
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = PopIIIEnergyBins[bin];
+      SS->RadiationSED[bin] = PopIIISEDFracBins[bin];
+    }
+  }
+  /*
+   * For the SMS spectrum I assume a blackbody with an effective 
+   * temperature of 6000K. 
+   * Luminosity is NOT in ergs/s but in photons/s 
+   */
+  else if (SS->ParticleClass == SMS) {
+    SS->RadiationLifetime = SmartStarSMSLifetime*3.154e7/TimeUnits; //Code Times
+    SS->LuminosityPerSolarMass = 1.4e51/500.0;
+    /* Ideally we call SLUG here. Hardcoded for now */
+    for(int bin = 0; bin < NUMRADIATIONBINS; bin++) {
+      SS->RadiationEnergyBins[bin] = SMSEnergyBins[bin];
+      SS->RadiationSED[bin] = SMSSEDFracBins[bin];
```

Comment:
Call SLUG before merging if possible.

Reply:
I don't think the SLUG PR has made it into Enzo yet.

Reply:
The SLUG PR was declined by the submitter and probably won't be incorporated. Removed the comment.